xbmc/platform/linux/PlatformLinux.h

```diff
@@ -23,7 +23,9 @@ class CPlatformLinux : public CPlatformPosix
   bool InitStageOne() override;
   void DeinitStageOne() override;
 
+#if defined(ADDONS_CONFIGURE_AT_STARTUP)
   bool IsConfigureAddonsAtStartupEnabled() override { return true; }
+#endif
```

Comment:
I don't like how the if-defery affects the API inheritance. I would prefer to not override conditionally, and encapsulate the if-defery in the function:

Header:

```c++
bool IsConfigureAddonsAtStartupEnabled() override;
```

Cpp:

```c++
bool CPlatformLinux::IsConfigureAddonsAtStartupEnabled()
{
#if defined(ADDONS_CONFIGURE_AT_STARTUP)
  return true;
#else
  return false;
#endif
}
```